### PR TITLE
[poco] re-enabling support for NetSSL with OpenSSL on Windows

### PR DESCRIPTION
**Pull request by jngrb**
Specify library name and version:  **poco/1.8.1 to 1.11.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

Fixing https://github.com/conan-io/conan-center-index/issues/6311 : I want to be able to consume poco (any version, currently available are Poco v1.8.1 to v1.11.0) with enable_netssl=True and enable_netssl_win=False to have SSL support using the OpenSSL on Windows (instead of Windows' TLS).

Before https://github.com/conan-io/conan-center-index/pull/4205 and https://github.com/conan-io/conan-center-index/commit/6ea152e89a6b2303739f59ebfc4d1e7e227bb8a3 were implemented, I could do this using:

- enable_netssl=True
- enable_netssl_win=False
- force_openssl=True

on Windows. As force_openssl is redundant information, this PR supports this use case with

- enable_netssl=True
- enable_netssl_win=False

only. Of course, setting both to False gives no SSL support (no NetSSL component) and WinTLS can be consumed by reversing these two flags. And setting both to True gives an ConanInvalidConfiguration exception.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Talkless** [comment]
I've hit on this issue when migrated packages to use new conan center repository. Would be pretty nice to have this fixed ASAP.

**Comment by Talkless** [comment]
Tested with my applications, builds and runs fine. @uilianries Please consider reviewing this.

**Review by mathbunnyru — CHANGES_REQUESTED** [review]
With this change, unfortunately, you will completely disable Windows CCI builds.

Why will this happen?
Both enable_netssl and enable_netssl_win are set True by default (and CCI always uses default config).
You don't do anything to these options on Windows and you will raise ConanInvalidConfiguration, which means it will *pass* on CI, but you will not be actually building this package for Windows.

**Comment by mathbunnyru** [comment]
I do agree with you, that I made a mistake, but we have to come up with a solution that is checked in the CCI, if it's possible.

**Comment by jngrb** [comment]
@mathbunnyru , I agree. I was unaware about the skipped CI tests until you told me.

I changed my PR by setting the default for the Windows-only option enable_netssl_win to false.
As the enable_netssl option's default is true (for any Platform), any consumer not specifying options will get the OpenSSL-supported NetSSL component.
For explicitly usage of the Windows Schannel SSP-supported NetSSL_win, the options have to be switched by the consumer.
Note that the Conan-Center-Index CI/CD will build and test the OpenSSL version on Windows.

Is that OK? Or should the CI/CD test the Schannel component on Windows?

(Note: by mistake, I wrote _WinTLS_ in my earlier comments but I meant Windows Schannel SSP).

**Review by mathbunnyru — APPROVED** [review]
I think it's ok to test openssl even on Windows.

**Comment by prince-chrismc** [comment]
> Note that the Conan-Center-Index CI/CD will build and test the OpenSSL version on Windows.
> 
> Is that OK? Or should the CI/CD test the Schannel component on Windows?

Totally acceptable, we have other recipes with many SSL options and most default to OpenSSL

**Review by prince-chrismc — APPROVED** [review]

File: recipes/poco/all/conanfile.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ class PocoConan(ConanFile):
         "PocoMongoDB": _PocoComponent("enable_mongodb", True, ("PocoNet", ), True),
         "PocoNet": _PocoComponent("enable_net", True, ("PocoFoundation", ), True),
         "PocoNetSSL": _PocoComponent("enable_netssl", True, ("PocoCrypto", "PocoUtil", "PocoNet", ), True),    # also external openssl
-        "PocoNetSSLWin": _PocoComponent("enable_netssl_win", True, ("PocoNet", "PocoUtil", ), True),
+        "PocoNetSSLWin": _PocoComponent("enable_netssl_win", False, ("PocoNet", "PocoUtil", ), True),
```
Comment (SSE4) [inline recipes/poco/all/conanfile.py:51]: for reviewers: it's not obvious from the diff itself what does True/False mean here
it's just a default option value according to the: 
https://github.com/conan-io/conan-center-index/blob/1ba9584da61a7119ac3f26d91ec0a6e7a9380dba/recipes/poco/all/conanfile.py#L32

**Review by SSE4 — APPROVED** [review]

**Pull request merged**